alibi_detect/od/pytorch/svm.py

```diff
@@ -0,0 +1,307 @@
+
+        Raises
+        ------
+        ValueError
+            If the kernel is not supported.
+        """
+        super().__init__(device=device)
+        self.n_components = n_components
+        self.kernel = kernel
+        self.sigma = sigma
+        self._init_kernel()
+
+    def _init_kernel(self) -> None:
+        """Initialize the Kernel."""
+
+        if isinstance(self.kernel, str):
+            if self.kernel not in ['rbf']:
+                raise ValueError(
+                    f'Currently only the rbf Kernel is supported for the SVM torch backend, got {self.kernel}.'
+                )
+            if self.kernel == 'rbf':
+                if self.sigma is not None:
+                    sigma = torch.tensor(self.sigma, device=self.device)
+                self.kernel = GaussianRBF(sigma=sigma)
+
+    def fit(  # type: ignore[override]
+        self,
+        x_ref: torch.Tensor,
+        nu: float = 0.5,
+        step_size_range: Tuple[float, float] = (1e-6, 1.0),
+        n_step_sizes: int = 16,
+        tol: float = 1e-6,
+        n_iter_no_change: int = 25,
+        max_iter: int = 1000,
+        verbose: int = 0,
+    ) -> Dict:
+        """Fit the SVM detector.
+
+        Parameters
+        ----------
+        x_ref
+            Training data.
+        nu
+            The proportion of the training data that should be considered outliers. Note that this does not necessarily
+            correspond to the false positive rate on test data, which is still defined when calling the
+            `infer_threshold()` method.
+        step_size_range
+            The range of values to be considered for the gradient descent step size at each iteration. This is
+            specified as a tuple of the form `(min_eta, max_eta)`.
+        n_step_sizes
+            The number of step sizes in the defined range to be tested for loss reduction. This many points are spaced
+            equidistantly along the range in log space.
+        tol
+            The decrease in loss required over the previous n_iter_no_change iterations in order to continue optimizing.
+        n_iter_no_change
+            The number of iterations over which the loss must decrease by `tol` in order for optimization to continue.
+        max_iter
+            The maximum number of optimization steps.
+        verbose
+            Verbosity level during training. ``0`` is silent, ``1`` a progress bar.
+
+        Returns
+        -------
+        Dictionary with fit results. The dictionary contains the following keys
+            - converged: bool indicating whether training converged.
+            - n_iter: number of iterations performed.
+            - lower_bound: loss lower bound.
+        """
+
+        self.nystroem = _Nystroem(
+            self.kernel,  # type: ignore[arg-type]
+            self.n_components
+        )
+
+        X_nys = self.nystroem.fit(x_ref).transform(x_ref)
```

Comment:
Noting that in `sklearn` backend the transformed data was called `x_ref`, although I think `X_nys` is clearer. Would suggest making the change in `sklearn` backend for consistency (and suggest sticking with lower-case convention, so `x_nys`).

Reply:
👍🏻